gcovr/html_generator.py

```diff
@@ -245,12 +248,14 @@ def print_html_report(covdata, output_file, options):
         data['ROWS'] = []
         currdir = os.getcwd()
         os.chdir(options.root_dir)
-        with io.open(data['FILENAME'], 'r', encoding=options.source_encoding,
-                     errors='replace') as INPUT:
-            for ctr, line in enumerate(INPUT, 1):
-                data['ROWS'].append(
-                    source_row(ctr, line.rstrip(), cdata.lines.get(ctr))
-                )
+        try:
+            with io.open(data['FILENAME'], 'r', encoding=options.source_encoding, errors='replace') as INPUT:
+                for ctr, line in enumerate(INPUT, 1):
+                    data['ROWS'].append(
+                        source_row(ctr, line.rstrip(), cdata.lines.get(ctr))
+                    )
+        except IOError as e:
+            logger.warn('File Generation failed: {filename}, reason={reason}', filename=data['FILENAME'], reason=repr(e))
```

Comment:
The HTML doesn't cotain data if source isn't found. Would it be better to use something like `!!! File not found !!!` for the first line and keeep the lines for the rest of the data empty?

Reply:
I'll change it to !!! File not found: {filename}, reason={reason}

Please check if this is what you suggested, thx!